ocs_ci/ocs/ui/base_ui.py

```diff
@@ -0,0 +1,172 @@
+        self.cnt_screenshot = 0
+
+    def do_click(self, by_locator, type=By.XPATH, timeout=30, screenshot=True):
+        """
+        Click on Button/link on OpenShift Console
+
+        by_locator(str): Command that tells Selenium IDE which
+        GUI element needs to operate on.
+        type(By): Set of supported locator strategies.
+        timeout(int): Looks for a web element repeatedly until timeout(sec) happens.
+        screenshot(bool): True- take screenshot, False-no take screenshot
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        element.click()
+        if screenshot:
+            self.take_screenshot()
+
+    def do_send_keys(self, by_locator, text, type=By.XPATH, timeout=30):
+        """
+        Send text to element on OpenShift Console
+
+        by_locator(str): Command that tells Selenium IDE which
+        GUI element needs to operate on.
+        text(str): Send text to element
+        type(By): Set of supported locator strategies
+        timeout(int): Looks for a web element repeatedly until timeout(sec) happens.
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        element.send_keys(text)
+
+    def is_expanded(self, by_locator, type=By.XPATH, timeout=30):
+        """
+        Check whether an element is in an expanded or collapsed state
+
+        Args:
+            by_locator(str): Command that tells Selenium IDE which
+            GUI element needs to operate on.
+            type(By): Set of supported locator strategies
+            timeout(int): Looks for a web element repeatedly until timeout(sec) happens.
+
+        return:
+            (str): "true"-element expanded, "false"-element collapsed
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        return element.get_attribute("aria-expanded")
+
+    def choose_expanded_mode(self, mode, by_locator, type):
+        """
+        Select the element mode (expanded or collapsed)
+
+        by_locator(str): Command that tells Selenium IDE which
+        GUI element needs to operate on.
+        type(By): Set of supported locator strategies
+        mode(str): Element mode (expanded or collapsed)
+
+        """
+        current_mode = self.is_expanded(by_locator=by_locator, type=type)
+        if mode != current_mode:
+            self.do_click(by_locator=by_locator, type=type)
+
+    def take_screenshot(self):
+        """
+        Take screenshot using python code
+
+        """
+        self.cnt_screenshot += 1
+        filename = os.path.join(
+            self.screenshots_folder, str(self.cnt_screenshot).zfill(3)
+        )
+        self.driver.save_screenshot(filename)
+
+
+def login_ui(browser, headless=True, chrome_type=ChromeType.CHROMIUM):
```

Comment:
how about having a class that will contain the various operations and can get the browser type to its initialization? this class can have the `login_ui` and `close_browser` functions as methods

Reply:
I use `login_ui` and `close_browser` on conftest file `def setup_ui`

Reply:
ack